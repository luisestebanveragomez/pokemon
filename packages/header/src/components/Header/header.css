.header {
    padding: 1rem;
    color: var(--white);
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 3;
}

.header__sectionImg-button {
    background: transparent;
    cursor: pointer;
}

.header__sectionImg-img {
    height: 50px;
}

.header__changeLanguage-button {
    background: transparent;
    font-size: 1.5rem;
    color: var(--white);
    display: flex;
    gap: 10px;
    cursor: pointer;
}

.header__changeLanguage-text {
    display: none;
}

.header__changeLanguage-arrow {
    transition: .3s;
    transform: rotate(0deg);
}

.header__changeLanguage-arrow.active {
    transition: .3s;
    transform: rotate(180deg);
}

.header__containerSelect {
    transition: .3s;
    right: 0;
    top: 28px;
    width: 70px;
    z-index: -1;
    opacity: 0;
}

.header__containerSelect.active {
    top: 38px;
    opacity: 1;
    z-index: 2;
}