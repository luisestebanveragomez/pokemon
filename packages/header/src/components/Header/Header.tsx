import { FC, useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";

import "./header.css";
import Pokemon from "../../assets/logo.png";
import { Select } from "../Select/Select";
import { Languages, languageStorage } from "../../constants/languages";
import { sendEventHelper } from "../../utils/helppers/sendEvent.helper";

export const Header: FC = () => {
  const navigate = useNavigate();
  const [showSelect, setShowSelect] = useState<boolean>(false);
  const [languageActive, setLanguageActive] = useState<string>(
    localStorage.getItem(languageStorage)
  );

  useEffect(() => {
    sendEventHelper(languageStorage, { language: languageActive });
    localStorage.setItem(languageStorage, languageActive);
  }, [languageActive]);

  const handleChangeLanguage = (el: string) => {
    setLanguageActive(el);
    handleShowSelect();
  };
  const handleShowSelect = () => setShowSelect((val) => !val);

  return (
    <header className="header">
      <section>
        <h1>
          <button
            className="header__sectionImg-button"
            onClick={() => navigate("/")}
          >
            <img
              src={Pokemon}
              className="header__sectionImg-img"
              alt="Pokémon"
              title="Logo Pokémon"
            />
          </button>
        </h1>
      </section>
      <aside className="relative">
        <button
          className="header__changeLanguage-button"
          onClick={handleShowSelect}
          aria-label="Change Language"
        >
          <i className="icon-translate"></i>
          <i
            className={`icon-arrow-down header__changeLanguage-arrow ${
              showSelect ? "active" : ""
            }`}
          ></i>
        </button>
        <div
          className={`absolute header__containerSelect ${
            showSelect ? "active" : ""
          }`}
        >
          <Select
            languages={Languages}
            active={languageActive}
            handle={handleChangeLanguage}
          />
        </div>
      </aside>
    </header>
  );
};
