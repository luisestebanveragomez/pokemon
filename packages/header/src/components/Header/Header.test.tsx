import { fireEvent, render, screen } from "@testing-library/react";
import { Header } from "./Header";

jest.mock("react-router-dom", () => ({
  __esModule: true,
  useNavigate: () => (key: string) => key,
}));

describe("Header", () => {
  it("Should Render", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it("Should Be Show Sub Menu Language", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
    const button = screen.getByLabelText("Change Language");
    const logo = container.querySelector(".header__sectionImg-button");
    const icon = button.querySelector(".header__changeLanguage-arrow");
    fireEvent.click(button);
    fireEvent.click(logo);
    expect(icon.classList.contains("active")).toBeTruthy();
  });
});
