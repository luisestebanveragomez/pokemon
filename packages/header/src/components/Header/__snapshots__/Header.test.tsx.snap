// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should Be Show Sub Menu Language 1`] = `
<div>
  <header
    class="header"
  >
    <section>
      <h1>
        <button
          class="header__sectionImg-button"
        >
          <img
            alt="Pokémon"
            class="header__sectionImg-img"
            src="[object Object]"
          />
        </button>
      </h1>
    </section>
    <aside
      class="relative"
    >
      <button
        aria-label="Change Language"
        class="header__changeLanguage-button"
      >
        <i
          class="icon-translate"
        />
        <i
          class="icon-arrow-down header__changeLanguage-arrow "
        />
      </button>
      <div
        class="absolute header__containerSelect "
      >
        <ul
          class="select__ul"
        >
          <li
            class="select__ul-list"
          >
            <button
              class="select__ul-button"
            >
              <span />
              es
            </button>
          </li>
          <li
            class="select__ul-list"
          >
            <button
              class="select__ul-button"
            >
              <span />
              en
            </button>
          </li>
        </ul>
      </div>
    </aside>
  </header>
</div>
`;

exports[`Header Should Render 1`] = `
<div>
  <header
    class="header"
  >
    <section>
      <h1>
        <button
          class="header__sectionImg-button"
        >
          <img
            alt="Pokémon"
            class="header__sectionImg-img"
            src="[object Object]"
          />
        </button>
      </h1>
    </section>
    <aside
      class="relative"
    >
      <button
        aria-label="Change Language"
        class="header__changeLanguage-button"
      >
        <i
          class="icon-translate"
        />
        <i
          class="icon-arrow-down header__changeLanguage-arrow "
        />
      </button>
      <div
        class="absolute header__containerSelect "
      >
        <ul
          class="select__ul"
        >
          <li
            class="select__ul-list"
          >
            <button
              class="select__ul-button"
            >
              <span />
              es
            </button>
          </li>
          <li
            class="select__ul-list"
          >
            <button
              class="select__ul-button"
            >
              <span />
              en
            </button>
          </li>
        </ul>
      </div>
    </aside>
  </header>
</div>
`;
